scripts/screeps_rl_dashboard.py

```
@@ -679,7 +969,113 @@ def simulator_health(
+            source="training_report_experiment_card",
+            run_id=text_value(payload.get("reportId")),
+        )
+    return None
+
+
+def card_supply_blocker_marker_present(value: Any) -> bool:
+    text = normalized_key(canonical_json(value))
+    return any(marker in text for marker in CARD_SUPPLY_BLOCKER_MARKERS)
+
+
+def card_supply_blocker_value_active(value: Any) -> bool:
```

Ruff (0.15.12)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `value`

(ANN401)